plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -142,37 +141,39 @@ func (plugin *T128GraphQL) checkConfig() error {
 
 // Gather takes in an accumulator and adds the metrics that the Input gathers
 func (plugin *T128GraphQL) Gather(acc telegraf.Accumulator) error {
-
-	processedResponses := plugin.MakeRequest(acc)
-	if processedResponses != nil {
-		for _, processedResponse := range processedResponses {
-			acc.AddFields(
-				plugin.CollectorName,
-				processedResponse.Fields,
-				processedResponse.Tags,
-			)
-		}
+	if !plugin.RetryIfNotFound && plugin.endpointNotFound {
+		return nil
+	}
+	processedResponses, err := plugin.MakeRequest()
+	if err != nil && processedResponses == nil {
+		acc.AddError(err)
+		return nil
+	} else if err != nil && processedResponses != nil {
+		acc.AddError(err)
+	}
+	for _, processedResponse := range processedResponses {
+		acc.AddFields(
+			plugin.CollectorName,
+			processedResponse.Fields,
+			processedResponse.Tags,
+		)
 	}
 
 	return nil
 }
 
 // MakeRequest can be used by other plugins using GraphQL functionality
-func (plugin *T128GraphQL) MakeRequest(acc telegraf.Accumulator) []*ProcessedResponse {
-	if !plugin.RetryIfNotFound && plugin.endpointNotFound {
-		return nil
-	}
+func (plugin *T128GraphQL) MakeRequest() ([]*ProcessedResponse, error) {
```

Comment:
Can we return `[]error` instead of cramming them all together into a string? In particular, the behavior of the graphql input should not change with this refactor. All the tests should continue to pass without modification.